### PR TITLE
LPS-143120 Add External Reference Code in RemoteAppEntry

### PR DESCRIPTION
**Pull request by joseabelenda**

### CODE REVIEW

File: modules/apps/remote-app/remote-app-service/src/main/java/com/liferay/remote/app/service/impl/RemoteAppEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -195,6 +195,46 @@ public RemoteAppEntry addIFrameRemoteAppEntry(
 		return _startWorkflowInstance(userId, remoteAppEntry);
 	}
 
+	@Indexable(type = IndexableType.REINDEX)
+	@Override
+	public RemoteAppEntry addOrUpdateCustomElementRemoteAppEntry(
+			long userId, String customElementCSSURLs,
+			String customElementHTMLElementName, String customElementURLs,
+			String description, String externalReferenceCode,
```
Comment (javierdearcos) [inline modules/apps/remote-app/remote-app-service/src/main/java/com/liferay/remote/app/service/impl/RemoteAppEntryLocalServiceImpl.java:203]: So sorry... I missed this in my first review. The externalReferenceCode param is always the first parameter as it is an identifier of the resource. Brian wanted this way for all the entities that already have it

File: modules/apps/remote-app/remote-app-service/src/main/java/com/liferay/remote/app/internal/upgrade/v2_4_0/RemoteAppEntryUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		addExternalReferenceCodeColumn();
+
+		updateExternalReferenceCode();
+	}
+
+	protected void updateExternalReferenceCode() throws Exception {
+		StringBundler sb = new StringBundler(2);
+
+		sb.append("update RemoteAppEntry set externalReferenceCode = ");
+		sb.append("remoteAppEntryId where externalReferenceCode is null ");
```
Comment (javierdearcos) [inline modules/apps/remote-app/remote-app-service/src/main/java/com/liferay/remote/app/internal/upgrade/v2_4_0/RemoteAppEntryUpgradeProcess.java:49]: Don't know if this SQL Query will work considering that the remoteAppEntryId is a bigint and the externalReferenceCode a VarChar.

In other upgrade processes we used the CAST_TEXT function to convert it

**Review by javierdearcos — CHANGES_REQUESTED** [review]
The externalReferenceCode parameter should be the first parameter in Service and LocalService methods (sorry I missed this in my first review).

Check the upgrade process. Don't know if the update will work as the remoteAppEntryId and the externalReferenceCode have different types

File: modules/apps/remote-app/remote-app-service/src/main/java/com/liferay/remote/app/internal/upgrade/v2_4_0/RemoteAppEntryUpgradeProcess.java (hunk 2 above)
Comment (joseabelenda) [reply modules/apps/remote-app/remote-app-service/src/main/java/com/liferay/remote/app/internal/upgrade/v2_4_0/RemoteAppEntryUpgradeProcess.java]: Yes, it works. I tested before send PR.
But I preferred follow your suggestion and use CAST_TEXT to make it consistent with the pattern.

File: modules/apps/remote-app/remote-app-service/src/main/java/com/liferay/remote/app/service/impl/RemoteAppEntryLocalServiceImpl.java (hunk 1 above)
Comment (joseabelenda) [reply modules/apps/remote-app/remote-app-service/src/main/java/com/liferay/remote/app/service/impl/RemoteAppEntryLocalServiceImpl.java]: No problem!
I changed to be first argument.

**Comment by joseabelenda** [comment]
@javierdearcos thanks for your review.
I made required changes and pushed them.

**Review by javierdearcos — APPROVED** [review]
Great! Thank you for your patience with my reviews.

I hope it helps to pass the final review at the first time and get it merge soon

**Comment by javierdearcos** [comment]
ci:forward

**Pull request closed without merging**